dodiscover/metrics.py

```diff
@@ -0,0 +1,66 @@
+import networkx as nx
+import numpy as np
+from sklearn.metrics import confusion_matrix
+from sklearn.preprocessing import LabelBinarizer
+
+from ._protocol import Graph
+
+
+def confusion_matrix_networks(
+    true_graph: Graph,
+    pred_graph: Graph,
+):
+    """Compute the confusion matrix comparing a predicted graph from the true graph.
+
+    Converts the graphs into adjacency matrices, which are symmetric.
+
+    Parameters
+    ----------
+    true_graph : instance of causal graph
+        The true graph.
+    pred_graph : instance of causal graph
+        The predicted graph. The predicted graph and true graph must be
+        the same type.
+
+    Returns
+    -------
+    cm : np.ndarray of shape (2, 2)
+        The confusion matrix.
+
+    See Also
+    --------
+    sklearn.metrics.confusion_matrix
+    """
+    if set(true_graph.nodes) != set(pred_graph.nodes):
+        raise RuntimeError("Both nodes should match.")
+
+    # convert graphs to adjacency graph in networkx
+    true_graph = true_graph.to_undirected()
+    pred_graph = pred_graph.to_undirected()
```

Comment:
Why to do we only want to look at undirected edges? The correctness of the direction could also be measured?

Reply:
This function is meant solely to test correctness of adjacencies which is done commonly in addition to directionality. Whereas something like SHD would handle the case of directionality too. 

I can update the doc string to elaborate.

Reply:
Ok, but note that there (seems to be) multiple definitions of a "adjacency matrix". Here, you assume that `a_{i, j} = a_{j, i}` which does not necessarily need to be the case for a "adjacency matrix".

Reply:
I have updated the docstring.